src/rwlock.jl

```diff
@@ -95,6 +95,23 @@ function Base.lock(rw::ReadWriteLock)
     return
 end
 
+function Base.trylock(rw::ReadWriteLock)
```

Comment:
is this trying to readlock or write lock?

Reply:
Writelock, since `lock` and `unlock` are for writelock. The readlock equivalent would likely be a custom `tryreadlock` function.